opentelemetry/deployment/otelcol.yaml

```diff
@@ -0,0 +1,101 @@
+apiVersion: project.openshift.io/v1
+kind: Project
+metadata:
+  name: otel-collector-example
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: otel-collector-deployment
+  namespace: otel-collector-example
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  labels:
+    config.openshift.io/inject-trusted-cabundle: "true"
+  name: trusted-ca
+  namespace: otel-collector-example
+data: {}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: otel-collector
+  namespace: otel-collector-example
+rules:
+- apiGroups: [""]
+  resources: ["pods", "namespaces"]
```

Comment:
```suggestion
  resources: ["pods", "namespaces", "infrastructures"]
```

Reply:
@frzifus 
Why is this needed? The [k8sattributesprocessor documentation don't mention it](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/k8sattributesprocessor#role-based-access-control). Is the documentation wrong or is something related to OpenShift?

Reply:
Its not wrong, queried by the openshift resource detection and not documented.